### PR TITLE
fix(forms): Tests on undefined.

### PR DESCRIPTION
**Pull request by AnObstacle**
Fix:
* The minlength test on undefined test null.
* Add undefined tests to required and maxlength

**What kind of change does this PR introduce?** (check one with "x")
```
[x] Bugfix
```

**What is the new behavior?**
Fix and add tests.

**Does this PR introduce a breaking change?** (check one with "x")
```
[x] No
```

### CODE REVIEW

**Comment by laco0416** [comment]
`pattern` validator also has a wrong test on `undefined`.

File: modules/@angular/forms/test/validators_spec.ts [hunk 1]
```diff
@@ -43,6 +43,9 @@ export function main() {
       it('should error on null',
          () => { expect(Validators.required(new FormControl(null))).toEqual({'required': true}); });
 
+      it('should not error on undefined',
+         () => { expect(Validators.required(2)(new FormControl(undefined))).toBeNull(); });
```
Comment (vicb) [inline modules/@angular/forms/test/validators_spec.ts]: extra `(2)` here ?

**Review by vicb — CHANGES_REQUESTED** [review]
Please make sure the CI is green and squash your commits

**Comment by laco0416** [comment]
IMHO, this change is one of a refactoring, so `refactor(forms): ...`  is proper for better CHANGELOG.

**Comment by Mathou54** [comment]
It seems more like a test(forms) no?

**Comment by vicb** [comment]
yep `test(forms)` seems the most appropriate here

**Review by vicb — APPROVED** [review]

**Review by kara — APPROVED** [review]
LGTM

**Pull request closed without merging**